src/utils.rs

```diff
@@ -170,3 +175,22 @@ macro_rules! vec_of_strings {
         vec![$(String::from($str),)*] as Vec<String>
     });
 }
+
+// Create the directory ~/.config/ht/sessions/{domain} if not present.
+pub fn ensure_session_dir_exists(domain: &str) -> std::io::Result<PathBuf> {
+    let mut config_dir = match dirs::config_dir() {
+        None => panic!("couldn't get config directory"),
+        Some(dir) => dir,
+    };
+    config_dir.push("ht");
+    config_dir.push("sessions");
+    config_dir.push(domain);
+    fs::create_dir_all(&config_dir)?;
+    Ok(config_dir)
+}
+
+// Calculate an hash from session identifier to use as the session filename.
+pub fn session_filename(identifier: &str) -> String {
+    let hash = Blake2b::new().chain(identifier).finalize();
+    format!("{:x}", hash).get(0..10).unwrap().to_string()
```

Comment:
Why only `0..=9` instead of the entire hash?

Reply:
Honestly, it's just a matter of having a shorter filename while still being mostly collision-free, a bit like what docker is doing with the short ids of the running containers. I tried with the full hash and it was _really_ long when listing the sessions directory.